### PR TITLE
Storybook: Rework the intro page

### PR DESCRIPTION
**Pull request by JoaoSilvaGrafana**
**What is this feature?**

Update the intro page to be more in line with how Storybook works now and to make mentions of Saga in it.

**Why do we need this feature?**

So this page it up to date

**Who is this feature for?**

Users of storybook

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

To be merged only when we make the external announcement for Saga.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/Intro.mdx [hunk 1]
```diff
@@ -2,11 +2,11 @@ import { Meta } from '@storybook/blocks';
 
 <Meta title="Docs Overview/Intro" parameters={{ options: { isToolshown: false }, id: 1 }} />
 
-# Grafana design system
+# Grafana React Components
 
 > **@grafana/ui is currently in BETA**.
```
Comment (Clarity-89) [inline packages/grafana-ui/src/Intro.mdx:7]: Should we remove this part? I don't think it's in beta anymore.

**Review by Clarity-89 — APPROVED** [review]

File: packages/grafana-ui/src/Intro.mdx (hunk 1 above)
Comment (jackw) [reply packages/grafana-ui/src/Intro.mdx:7]: If so we should also remove it from the [README](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/README.md).

File: packages/grafana-ui/src/Intro.mdx [hunk 2]
```diff
@@ -17,23 +17,19 @@ Grafana Labs started @grafana/ui to make contributing to Grafana as easy as poss
...
+When we notice that we need to change something, we determine together what the change should be, then we put the change in place and communicate it publicly. Developers and designers create and improve @grafana/ui together. Throughout the process, we strive to involve you and meet your needs. We are looking forward to discussing your design and improvement ideas, find out how you can help in the [Contribution Guidelines](https://grafana.com/developers/saga/contributing).
 
 ## Get started
 
 - **Explore UI components**
-  Click on any of the component in the sidebar to see how they look, and how to configure them. Each component pages contains documentation of its properties, as well as code examples for how to use it.
+  Click on any of the component in the sidebar to see how they look, and how to configure them. Each component contains documentation of its properties, as well as code examples for how to use it.
```
Comment (jackw) [inline packages/grafana-ui/src/Intro.mdx:27]:
```suggestion
  Click on any of the components in the sidebar to see how they look, and how to configure them. Each component contains documentation of its properties, as well as code examples for how to use it.
```

File: packages/grafana-ui/src/Intro.mdx (hunk 1 above)
Comment (JoaoSilvaGrafana) [reply packages/grafana-ui/src/Intro.mdx:7]: Yeah I think we should remove it from both of these 👍

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Pull request merged**